src/ImageSharp/Processing/Processors/Transforms/AffineProcessor.cs

```diff
@@ -0,0 +1,344 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.Helpers;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.MetaData.Profiles.Exif;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Processors
+{
+    /// <summary>
+    /// Provides the base methods to perform affine transforms on an image.
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal abstract class AffineProcessor<TPixel> : CloningImageProcessor<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        private Rectangle targetRectangle;
```

Comment:
The XY values of `targetRectangle` are actually never used!

Reply:
Using a `Rectangle` allows us to use `Rectangle.Intersect` though and easily handle negative values. 

I'm not really following what your expectations are (I must be feeling slow), would it be possible for you to diagram them out? We should see what other libraries do also.

Reply:
No worries, it's easy to tell them in a few images. Will do it in the next few days.
The implementation should be also easy, could be done within an hour.

Reply:
👍 Great news, thanks!